### PR TITLE
Adding support for InfluxDB 0.9.3

### PR DESCRIPTION
**Pull request by eliperelman**
InfluxDB 0.9.3 released a breaking change in its API for writing integers, so this is the most important change here. Summary of the other changes:
- Default "number" as `FLOAT64` to be consistent with InfluxDB number writing behavior. Remove trailing 0 from floats as all numbers are considered floats unless trailed with "i".
- Write `INT64` to line with a trailing "i". Numbers are never written as integers unless specified in the value, e.g. `new Value(10, influent.type.INT64)`
- Allow writing timestamps using strings. When using the nanosecond precision, some really large numbers in JS are more easily represented as strings.
- Bumped minor version number to represent breaking change. I know this is pre-1.0.0, so should this go 1.0.0?
- Added npm target for running gulp without global dependency.
- Added `--recursive` to mocha to support running `npm test` to test everything.
- Updated and added new tests to support changes.
- Update docs to reflect changes, better readability on GitHub

### CODE REVIEW

File: lib/serializer/line.js [hunk 1]
```diff
@@ -61,10 +61,6 @@ asFloat = function(obj) {
 
     str = obj.toString();
 
-    if (str.indexOf(".") == -1) {
-        return str + ".0";
-    }
-
     return str;
```
Comment (gobwas) [inline lib/serializer/line.js]: Could it be shortly `return obj.toString();`? =)

File: package.json [hunk 2]
```diff
@@ -45,7 +46,8 @@
     "vinyl-source-stream": "^1.1.0"
   },
   "scripts": {
-    "test": "mocha ./test/unit"
+    "test": "mocha --recursive ./test/unit",
+    "gulp": "gulp"
```
Comment (gobwas) [inline package.json]: Lets name it `build`, instead of concrete implementation =)
`"build": "gulp"` I mean )

File: test/unit/serializer/line.js [hunk 3]
```diff
@@ -129,6 +140,48 @@ describe("LineSerializer", function() {
...
+            measurement.setTimestamp(stamp);
+            result = instance.serialize(measurement);
+
+            // then
+            return result.should.become("key a=0 " + stamp.getTime().toString());
+        });
+
     });
 
 });
```
Comment (gobwas) [inline test/unit/serializer/line.js]: Please, add new line here )

**Comment by gobwas** [comment]
Hi @eliperelman!
Thank you so much for your work! :beers: 
I agree almost with all of your changes.

Here my questions:

> Allow writing timestamps using strings. When using the nanosecond precision, some really large numbers in JS are more easily represented as strings.

Shall we really move it into the `Measurement` model? I mean, may be we could leave it in the `DecoratorClient`? He is here exactly for easily appending values?

My logic is `Measurement` is storage for typed values, when the `DecoratorClient` is for dynamic typed values.

> Bumped minor version number to represent breaking change. I know this is pre-1.0.0, so should this go 1.0.0?

The `0.3.0` is ok, I think we could bump it to `1.0.0` on InfluxDB stable release?

Also, I've leaved little comments on your pr, could you fix them?

Thanks again!

**Comment by eliperelman** [comment]
@gobwas I've got most of the changes done, I'll commit them shortly.

> Shall we really move it into the Measurement model? I mean, may be we could leave it in the DecoratorClient? He is here exactly for easily appending values?
> My logic is Measurement is storage for typed values, when the DecoratorClient is for dynamic typed values.

The reason I did this was to accommodate the test, specifically in the line serializer, but it shouldn't be hard to move the test logic to the client decorator and then switch the logic back out of measurement. I should have that ready shortly!

**Comment by eliperelman** [comment]
@gobwas ok, should be updated. :+1:

**Comment by eliperelman** [comment]
@gobwas I just noticed when calling `influent.createClient`, that it rejects with an error about accepting an object. It appears to stem from https://github.com/gobwas/influent/blob/master/index.js#L66, since nothing is passed into the `DecoratorClient`. Does that seem suspect to you?

**Comment by gobwas** [comment]
@eliperelman yeah, looks like this is my mistake after adding options to the `Client` constructor. Will fix it during tomorrow it is too late in Moscow now :night_with_stars: , thanks!

**Comment by gobwas** [comment]
@eliperelman also, I just reviewed your changes about timestamp - I thought we will keep `number` in `Measurement` and `DecoratorClient` will just cast other types/objects to the `number`. Could you explain me again, why it is better to use `string` there? Do I understand it right, that nanoseconds precision is like `Date.now() * 10e6` ?

**Comment by gobwas** [comment]
@eliperelman Just fixed bug with `influent.createClient`, and now the PR conflicts. But it was necessary to release `0.2.3`. When we will be done on conversation above - i will fix this =)

**Comment by eliperelman** [comment]
@gobwas I rebased this PR, so the conflicts should be good to go.

Regarding the number/string issue for timestamp, here is a real use case from a project of mine: I would like to use a timestamp with nanosecond precision:

``` js
var timestamp = 1441319788996000001;

console.log(timestamp); // 1441319788996000000
```

Notice how the number had its last digit changed from a `1` to a `0`? This is the problem I can't avoid by using numbers. By changing timestamp to accept strings, we can accept the epochs that were presented before, and also representations of large numbers as strings. They are all written to the line in the same way. Make sense?

**Comment by gobwas** [comment]
@eliperelman yes, now I totally understand your opinion. 100% right =)
Thank you again for your work. 

P.s. How do you think, may be it is a good idea to make in future `Timestamp` entity that will hold the value? And, may be, some methods like `Timestamp.fromDate`, `Timestamp.fromString`? Cause it is a little bit strange to hold timestamp value as string, that at first glance should be a numeric..

**Comment by gobwas** [comment]
Merged :beers: Thanks!

**Comment by eliperelman** [comment]
Thanks for the merge! Could you also publish to npm?

As far as timestamp, while the concept seems to make sense, I'm not sure if it would be stable. If you wanted to `JSON.stringify` the entity, or use a string, wouldn't you still need to store as string? Also important is the distinction that these are **InfluxDB timestamps**, not JavaScript ones. Maybe it would be better to denote that in the docs so there isn't that confusion?

**Comment by gobwas** [comment]
Yeah, will publish it in 5 minutes. 
I've also added an assertion in `setTimestamp` for `_.isNumericString`. :sunglasses:

**Pull request merged**